Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift

```diff
@@ -0,0 +1,276 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  OfflineStoreKitIntegrationTests.swift
+//
+//  Created by Nacho Soto on 5/17/23.
+
+import Nimble
+@testable import RevenueCat
+import StoreKit
+import XCTest
+
+// swiftlint:disable type_name
+
+class BaseOfflineStoreKitIntegrationTests: BaseStoreKitIntegrationTests {
+
+    fileprivate var serverIsDown: Bool = false
+    override var forceServerErrors: Bool { return self.serverIsDown }
+
+    override func setUp() async throws {
+        self.serverIsDown = false
+        try await super.setUp()
+
+        await self.waitForPendingCustomerInfoRequests()
+    }
+
+}
+
+class OfflineStoreKit2IntegrationTests: OfflineStoreKit1IntegrationTests {
+
+    override class var storeKit2Setting: StoreKit2Setting { return .enabledForCompatibleDevices }
+
+}
+
+class OfflineStoreKit1IntegrationTests: BaseOfflineStoreKitIntegrationTests {
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) {
+            try await self.ensureEntitlementMappingIsAvailable()
+        }
+    }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testOfflineCustomerInfoFailsIfNoEntitlementMapping() async throws {
```

Comment:
Oh wait, I think this test should be removed since it's in its own class

Reply:
Thanks for catching it.